ui/imports/shared/views/NetworkCardsComponent.qml

```diff
@@ -0,0 +1,267 @@
+            Layout.alignment: Qt.AlignLeft | Qt.AlignTop
+            spacing: 12
+            StatusBaseText {
+                font.pixelSize: 10
+                color: Theme.palette.baseColor1
+                text: qsTr("Your Balances").toUpperCase()
+            }
+            Repeater {
+                model: networkCardsComponent.layer1Networks
+                StatusCard {
+                    id: layer1FromNetwork
+                    property var tokenBalanceOnChain: Utils.toLocaleString(parseFloat(d.getBalance(model.chainId)).toFixed(4), locale, {"currency": true})
+                    property var hasGas: assets.hasGas(model.chainId, requiredGasInEth + parseFloat(amountToSend))
+                    primaryText: model.chainName
+                    secondaryText: parseFloat(tokenBalanceOnChain) === 0 && amountToSend !== 0 ? qsTr("No Balance") : !hasGas ? qsTr("No Gas") :
+                                                                                                                                (selectedNetwork && selectedNetwork.chainName === model.chainName) ? amountToSend: 0
+                    tertiaryText: qsTr("BALANCE: ") + tokenBalanceOnChain
+                    state: tokenBalanceOnChain === 0 || !hasGas ? "unavailable" : networkCardsComponent.errorMode ? "error" : "default"
+                    cardIcon.source: Style.png(model.iconUrl)
+                    disabledText: "Disabled"
+                    advancedMode: networkCardsComponent.customMode
+                    advancedInputText: (selectedNetwork && selectedNetwork.chainName === model.chainName) ? amountToSend: 0
+                    Component.onCompleted: {
+                        disabled = store.checkIfDisabledByUser(model.chainId)
+                        if(selectedNetwork && selectedNetwork.chainName === model.chainName)
+                            d.selectedFromNetwork = this
+                    }
+                    Connections {
+                        target: networkCardsComponent
+                        onSelectedNetworkChanged: {
+                            if(selectedNetwork.chainName === model.chainName) {
+                                d.selectedFromNetwork = layer1FromNetwork
+                            }
+                        }
+                    }
+                    onClicked: {
+                        store.addRemoveDisabledChain(suggestedRoutes, model.chainId, disabled)
+                        reCalculateSuggestedRoute(store.disabledNetworks)
+                    }
+                    onAdvancedInputTextChanged: {
+                        if(selectedNetwork && selectedNetwork.chainName === model.chainName) {
+                            d.customAmountToSend = isNaN(parseFloat(advancedInputText)) ? 0 : parseFloat(advancedInputText)
+                        }
+                    }
+                }
+            }
+            Repeater {
+                model: networkCardsComponent.layer2Networks
```

Comment:
why not having a allNetworks variable?

Reply:
hmm if you mean introducing a new variable? I think its a good idea!